Submit form: Support commas in tag input, mark required fields

lmcnulty opened the pull request:
Implements all of the changes named in #1112.

- In tag inputs, a comma now begins a new tag, and a pasted comma-separated list get split into its components.
- Adds an asterisk before required fields.
- Hides the entity fields when .
- Sets the max description length to 500.

cesarvarela commented on site/gatsby-site/src/components/forms/TextInputGroup.js:
```diff
@@ -14,31 +14,40 @@ const TextInputGroup = ({
+}) => {
+  const [optional, setOptional] = useState(true);
+
+  useEffect(async () => {
+    setOptional(await schema.fields[name].isValid(undefined));
+  }, []);
```
Not sure I follow this 🤔

lmcnulty replied:
`schema` contains the full `yup` schema. We descend into the field matching the current input group, then we test if the field is optional according to the schema by seeing if `isValid` returns `true` given `undefined`. For some reason, `isValid` is asynchronous, so we need to wrap it in a `useEffect`.

cesarvarela replied:
Oh my! There must be a better way to do this, but let's go with it for now. Forms need a rework anyway, including moving them to flow bite/tailwind.

cesarvarela reviewed (commented):
One question before merging

cesarvarela commented:
@lmcnulty since you are at it:

`author` and `date_downloaded` show an asterisk but no invalid feedback message:

![image](https://user-images.githubusercontent.com/1172479/191374638-a82e2790-baff-445b-9b51-ed8d65deef53.png)

*This is by pressing `Submit`)

lmcnulty commented:
@cesarvarela I fixed the validation for tag inputs. The "Date Downloaded" field doesn't show an error because it defaults to the current day as of the last PR.

<img src="https://user-images.githubusercontent.com/25443411/191596932-6740639d-5286-414b-a131-e5aec1e39d13.png" width="500"/>

While I was there I also fixed the corner weirdness from #943.
<table>
  <tr><th>Before</th><th>After</th>
  <tr>
    <td><img width="500" src="https://user-images.githubusercontent.com/25443411/191597287-2e37b37a-d245-406e-8cb0-379d8ea2b421.png"/></td>
    <td><img width="500" src="https://user-images.githubusercontent.com/25443411/191597475-37db7a98-911b-426c-b19a-2399c941e651.png"/></td>
  </tr>
</table>

cesarvarela commented:
@lmcnulty please merge staging, tests should pass now
